File: bindings/java-jni/src/main/java/org/maplibre/nativejni/render/MetalOwnedTextureDescriptor.java
package org.maplibre.nativejni.render;

import java.util.Objects;

/** Mutable descriptor for Metal session-owned texture render targets. */
public final class MetalOwnedTextureDescriptor {
  private RenderTargetExtent extent = new RenderTargetExtent();
  private MetalContextDescriptor context = new MetalContextDescriptor();

  public RenderTargetExtent extent() {
    return extent;
  }

  public MetalOwnedTextureDescriptor extent(RenderTargetExtent extent) {
    this.extent = Objects.requireNonNull(extent, "extent");
    return this;
  }

  public MetalContextDescriptor context() {
    return context;
  }

  public MetalOwnedTextureDescriptor context(MetalContextDescriptor context) {
    this.context = Objects.requireNonNull(context, "context");
    return this;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/render/MetalOwnedTextureDescriptor.kt
package org.maplibre.nativeffi.render

/** Mutable descriptor for Metal session-owned texture render targets. */
public class MetalOwnedTextureDescriptor(
  extent: RenderTargetExtent = RenderTargetExtent(),
  context: MetalContextDescriptor = MetalContextDescriptor(),
) {
  public var extent: RenderTargetExtent = extent

  public var context: MetalContextDescriptor = context
}
